dbt-tests-adapter/dbt/tests/adapter/incremental/test_incremental_microbatch.py

```diff
@@ -0,0 +1,84 @@
+import os
+from unittest import mock
+
+import pytest
+from freezegun import freeze_time
+
+from dbt.tests.util import relation_from_name, run_dbt
+
+_input_model_sql = """
+{{ config(materialized='table', event_time='event_time') }}
+select 1 as id, TIMESTAMP '2020-01-01 00:00:00-0' as event_time
+union all
+select 2 as id, TIMESTAMP '2020-01-02 00:00:00-0' as event_time
+union all
+select 3 as id, TIMESTAMP '2020-01-03 00:00:00-0' as event_time
+"""
+
+_microbatch_model_sql = """
+{{ config(materialized='incremental', incremental_strategy='microbatch', unique_key='id', event_time='event_time', batch_size='day') }}
+select * from {{ ref('input_model') }}
+"""
+
+
+class BaseMicrobatch:
+    @pytest.fixture(scope="class")
+    def microbatch_model_sql(self):
+        return _microbatch_model_sql
+
+    @pytest.fixture(scope="class")
+    def input_model_sql(self):
+        return _input_model_sql
+
+    @pytest.fixture(scope="class")
+    def models(self, microbatch_model_sql, input_model_sql):
+        return {
+            "input_model.sql": input_model_sql,
+            "microbatch_model.sql": microbatch_model_sql,
+        }
+
+    def assert_row_count(self, project, relation_name: str, expected_row_count: int):
+        relation = relation_from_name(project.adapter, relation_name)
+        result = project.run_sql(f"select count(*) as num_rows from {relation}", fetch="one")
+
+        if result[0] != expected_row_count:
+            # running show for debugging
+            run_dbt(["show", "--inline", f"select * from {relation}"])
+
+            assert result[0] == expected_row_count
+
+    @mock.patch.dict(os.environ, {"DBT_EXPERIMENTAL_MICROBATCH": "True"})
+    def test_run_with_event_time(self, project):
+        # initial run -- backfills all data
+        with freeze_time("2020-01-03 13:57:00"):
+            run_dbt(["run"])
+        self.assert_row_count(project, "microbatch_model", 3)
+
+        # our partition grain is "day" so running the same day without new data should produce the same results
+        with freeze_time("2020-01-03 14:57:00"):
+            run_dbt(["run"])
+        self.assert_row_count(project, "microbatch_model", 3)
+
+        # add next two days of data
+        test_schema_relation = project.adapter.Relation.create(
+            database=project.database, schema=project.test_schema
+        )
+        project.run_sql(
+            f"insert into {test_schema_relation}.input_model(id, event_time) values (4, TIMESTAMP '2020-01-04 00:00:00-0'), (5, TIMESTAMP '2020-01-05 00:00:00-0')"
```

Comment:
I'm surprised there's no syntax error from the lack of a space between `input_model` and `(id, event_time)`. Maybe it was just personal preference that I always put a space there. I assume this runs successfully in `dbt-postgres` though? I'd also consider making this a separate method just in case an adapter maintainer need to overwrite it because the `INSERT` syntax is a bit different. Then they wouldn't need to overwrite the whole test.